docs/apis/subsystems/external/writing-a-service.md

```diff
@@ -316,10 +316,20 @@ Because some web service protocols are strict about the number and types of argu
 ```php
 public static function get_biscuit_parameters() {
     return new external_function_parameters([
-        'chocolatechips' => new external_value(PARAM_BOOL, PARAM_REQUIRED),
-        'glutenfree' => new external_value(PARAM_BOOL, PARAM_DEFAULT, false),
+        'chocolatechips' => new external_value(
+            PARAM_BOOL,
+            'if biscuit contains chocolate chips',
+            VALUE_REQUIRED,
+            VALUE_DEFAULT
```

Comment:
Ah - what is the `VALUE_DEFAULT` here? Fourth param is the _actual_ default value, so i think this was me getting everything confused and can come out.

As I say, why a move to named args is a really good thing going forwards.

Reply:
yes, overlooked that one. Removed the VALUE_DEFAULT.